src/zoid/buttons/util.js

```diff
@@ -389,10 +399,18 @@ export const getModal: () => Object = memoize(() => {
       return new Promise((resolve) => {
         const script = document.createElement("script");
         script.src = modalBundleUrl();
+        script.setAttribute("data-namespace", namespace);
+        script.setAttribute("data-pp-namespace", namespace);
         document.body?.appendChild(script);
         script.addEventListener("load", () => {
```

Comment:
I believe we can leverage an `"error"` event to capture failures from the script (e.g. network error) [according to MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/error_event) so we can use that to round out this promise based return value.